### PR TITLE
New graphs

### PR DESCRIPTION
**Pull request by Xesme**
This pull request implements the following changes on branch **master**: 
- [x] Adds a new tab called graphs within admin screens.
- [x] Adds a pie chart featuring current rate of successful and failed jobs.
- [x] Adds a scatter chart featuring the number of successful jobs for each month with filterable year. 
- [x] Add a bar graph featuring number of successful and failed jobs for each month by filtered year.

### CODE REVIEW

File: app/src/scripts/admin/admin.store.js [hunk 1]
```diff
@@ -192,6 +194,66 @@ let UserStore = Reflux.createStore({
...
+      FAILED: [],
+      SUCCEEDED: [],
+    }
+
+    for (let job of uploaded) {
+      let status = job.data.job.status
+      // convert date
+      let dateTime = new Date(job.date).toString()
+      let explode = dateTime.split(' ')
+      let year = explode[3]
```
Comment (nellh) [inline app/src/scripts/admin/admin.store.js]: This can just be `const dateTime = new Date(...).year`

File: app/src/scripts/admin/admin.store.js [hunk 2]
```diff
@@ -535,6 +597,9 @@ let UserStore = Reflux.createStore({
       let eventLogs = res.body
       this.update({ eventLogs }, () => {
         this.searchLogs('')
+        if (eventLogs.length) {
```
Comment (nellh) [inline app/src/scripts/admin/admin.store.js:665]: A guard like this can just be part of the filterLogs function, it looks like you duplicate it in different areas.

File: app/package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
   },
   "scripts": {
     "start":
-      "webpack-dev-server --config webpack.dev.js --progress --watch --color",
+      "webpack-dev-server --config webpack.dev.js --watch-poll --progress --watch --color",
```
Comment (nellh) [inline app/package.json]: I think you didn't mean to commit this. You can commit part of a file with `git add -p <filename>`

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,23 @@
 config.env
 .env
 
+.vscode/settings.json
+app/src/css/main.css.map
```
Comment (nellh) [inline .gitignore]: What are these css map files for?

File: app/src/scripts/admin/admin.graphs.jsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+  _handleSelect(e) {
+    this.setState({
+      year: e.target.value,
+    })
+    this._filterData(e.target.value)
+  }
+
+  _filterData(year) {
+    let entries = this.state.entries
```
Comment (nellh) [inline app/src/scripts/admin/admin.graphs.jsx]: This should be done in a new array or a copy of this.state.entries and applied with setState to prevent state changes that do not call lifecycle methods.

File: app/src/scripts/admin/admin.store.js [hunk 6]
```diff
@@ -192,6 +194,66 @@ let UserStore = Reflux.createStore({
...
 
+  // ** Filters Eventlog to smaller logs based on job status *
+  // uploading contains all jobs
+  filterLogs() {
+    let eventLogs = this.data.eventLogs
+    // let failedLogs = []
+    // let successLogs = []
+    let uploadedLogs = []
+    if (!eventLogs.length) {
+      this.getEventLogs()
```
Comment (nellh) [inline app/src/scripts/admin/admin.store.js]: Why is this necessary? I think this action shouldn't result in a call to getEventLogs, a caller should emit both actions if that is expected.

File: app/src/scripts/admin/admin.store.js [hunk 7]
```diff
@@ -192,6 +194,66 @@ let UserStore = Reflux.createStore({
...
+      } else if (entries[status][year]) {
+        entries[status][year].push({ dateTime: dateTime, log: job })
+        // push to mediary array
+      }
+      if (!yearArr.includes(year)) {
+        yearArr.push(year)
+      }
+    }
+    activityLogs = entries
+    this.update({ activityLogs })
```
Comment (nellh) [inline app/src/scripts/admin/admin.store.js]: This is overloading one kind of state with another. It should be its own state variable in the store.

File: server/libs/orcid.js [hunk 8]
```diff
@@ -3,7 +3,6 @@ import xmldoc from 'xmldoc'
 import config from '../config'
```
Comment (nellh) [inline server/libs/orcid.js]: It doesn't look like you changed anything in this file, try to avoid adding files that don't have logical changes. If you do want to update whitespace, it should be in a commit saying so.

**Review by nellh — CHANGES_REQUESTED** [review]
This looks pretty close. I made some implementation comments and asked some questions.

One bug I'm seeing is refreshing the page leads to a crash.

**Comment by nellh** [comment]
One more thing, maybe we should call it 'job stats' or make the tab 'stats' but title the page 'job statistics' so it's clearer what it is graphs of?

**Comment by Xesme** [comment]
@nellh sounds great! It looks like I accidentally committed a few things I shouldn't have and there is some unnecessary code. I will go through today and clean all up these files. And I will update the table and the url(route) to be clearer. I wasn't sure what to call it in the beginning and it just stayed the way.  :)  

Looks like the css.map files got added to my .gitignore because of sourceTree. I will remove them. 

Also, I am not exactly sure why the crash is happening. Do you see anything that is obviously causing the crash? My console doesn't say anything to me when it white screens. I was hoping it had something to do with code splitting. 

Thank you for the feedback!

**Comment by nellh** [comment]
Here's the traceback and line affected.

```
Uncaught TypeError: Cannot read property '2018' of undefined
    at Progresssion.render (admin.graphs.jsx:56)
    at finishClassComponent (react-dom.development.js:7873)
    at updateClassComponent (react-dom.development.js:7850)
    at beginWork (react-dom.development.js:8225)
    at performUnitOfWork (react-dom.development.js:10224)
    at workLoop (react-dom.development.js:10288)
    at HTMLUnknownElement.callCallback (react-dom.development.js:542)
    at HTMLUnknownElement.wrapped (raven.js:360)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:581)
    at invokeGuardedCallback (react-dom.development.js:438)
    at renderRoot (react-dom.development.js:10366)
    at performWorkOnRoot (react-dom.development.js:11014)
    at performWork (react-dom.development.js:10967)
    at requestWork (react-dom.development.js:10878)
    at scheduleWorkImpl (react-dom.development.js:10732)
    at scheduleWork (react-dom.development.js:10689)
    at scheduleTopLevelUpdate (react-dom.development.js:11193)
    at Object.updateContainer (react-dom.development.js:11231)
    at react-dom.development.js:15226
    at Object.unbatchedUpdates (react-dom.development.js:11102)
    at renderSubtreeIntoContainer (react-dom.development.js:15225)
    at Object.render (react-dom.development.js:15290)
    at Object.<anonymous> (client.jsx:27)
    at __webpack_require__ (bootstrap fc54aa0c31e4987599c5:19)
    at Object.<anonymous> (app.bundle.js:116414)
    at __webpack_require__ (bootstrap fc54aa0c31e4987599c5:19)
    at Object.CRN_SERVER_URL (bootstrap fc54aa0c31e4987599c5:62)
    at bootstrap fc54aa0c31e4987599c5:62
```

```
let failures = this.state.admin.activityLogs.FAILED[year].length
```

**Comment by Xesme** [comment]
@nell I believe this should be ready for another review. :)

**Review by nellh — APPROVED** [review]
